internal/store/digest.go

```diff
@@ -32,21 +32,26 @@ func FromString(s string) (ID, error) {
 	if ok, _ := regexp.MatchString("[a-z0-9]{64}", s); !ok {
 		return ID{}, fmt.Errorf("could not parse '%s' as a valid reference", s)
 	}
-	return ID{s}, nil
+	digest := digest.NewDigestFromEncoded(digest.SHA256, s)
+	return ID{digest}, nil
 }
 
 func FromBundle(bndle *bundle.Bundle) (ID, error) {
 	digest, err := ComputeDigest(bndle)
-	return ID{digest.Encoded()}, err
+	return ID{digest}, err
 }
 
 // ID is an unique identifier for docker app image bundle, implementing reference.Reference
 type ID struct {
```

Comment:
Shouldn't this struct and all associated methods (`String`, `Digest` but also `FromString` and `FromBundle`) be in a `id.go` file instead of a `digest.go`?

Reply:
Naming always is a problem :)
But remember those _are_ digests

Reply:
yep, but it's a question of consistency between file name and code.